backend/app.py

```diff
@@ -208,5 +208,110 @@ def get_by_keyword():
         return json.dumps({'success': False, 'error': Error.MISSING_KEYWORD_UPC.value})
 
 
+@app.route('/optimal_store', methods=['POST'])
+def get_optimal_store():
+    """
+        Body: {"single_store": boolean, single or multiple stores, "items": [upc:String]}
+        Response:   {"success": True or False,
+                    "error": error description,
+                    "optimal_prices":
+                            [{"store": store,
+                              "upcs": list of upcs,
+                              "price": total price of items associated with store}]
+                    }
+    """
+    data = request.get_json(force=True)
+    if data is None:
+        return json.dumps({'success': False, 'error': 'Could not parse JSON body', 'optimal_prices': None})
+
+    item_list = []
+    items = data['items']
+    if items is None:
+        return json.dumps({'success': False, 'error': 'No UPCs provided', 'optimal_prices': None})
+    for upc in items:
+        item_list.append(model.Item.objects(upc=upc).to_json())
```

Comment:
If the upc is not in our db this will append an empty array to item_list. This will then cause an IndexError at line 237. Solution may be just to not append the empty array and send an error/warning in the response

Reply:
Done